newIDE/app/src/UI/Search/UseSearchItem.js

```diff
@@ -4,18 +4,39 @@ import { type ChosenCategory } from './FiltersChooser';
 import shuffle from 'lodash/shuffle';
 import SearchApi from 'js-worker-search';
 
+export interface SearchFilter<SearchItem> {
+  isSatisfiedBy(searchItem: SearchItem): boolean;
+}
+
+export class TagSearchFilter<SearchItem: { tags: Array<string> }>
+  implements SearchFilter<SearchItem> {
+  tags: Set<string>;
+
+  constructor(tags: Set<string> = new Set()) {
+    this.tags = tags;
+  }
+
+  isSatisfiedBy(searchItem: SearchItem): boolean {
+    return (
+      this.tags.size === 0 || searchItem.tags.some(tag => this.tags.has(tag))
+    );
+  }
+}
+
 /**
  * Filter a list of items according to the chosen category
  * and the chosen filters.
  */
 export const filterSearchItems = <SearchItem: { tags: Array<string> }>(
   searchItems: ?Array<SearchItem>,
   chosenCategory: ?ChosenCategory,
-  chosenFilters: Set<string>
+  chosenFilters: Set<string>,
+  searchFilters?: Array<SearchFilter<SearchItem>>
 ): ?Array<SearchItem> => {
   if (!searchItems) return null;
 
   const startTime = performance.now();
+  // TODO do only one call to filter for efficiency.
```

Comment:
is this to avoid multiple calls when selecting multiple filters quickly?

Reply:
oh right, I misread the line for "filter for efficiency" instead of "to .filter() for efficiency" :)